src/services/zapper.ts

```diff
@@ -31,9 +31,14 @@ export class ZapperService extends Service {
       .then(handleHttpError)
       .then(res => res.json());
     const network = Chains[this.chainId] ?? "ethereum";
-    return tokens.map(
-      (token: Record<string, string>): Token => {
-        const address = token.address === ZeroAddress ? EthAddress : getAddress(String(token.address));
+
+    const validTokens: ZapperToken[] = tokens.filter((token: ZapperToken) =>
+      Object.values(token).every(attr => !!attr)
+    );
```

Comment:
We were assuming that the tokens had all the -- what I now called -- `ZapperToken` attributes, i.e.:

```typescript
export interface ZapperToken {
  address: string;
  hide: string;
  symbol: string;
  decimals: string;
  price: string;
}
```

Wondering if we should be mapping the zapper token to Token if some of them are missing?

Reply:
Digging a bit deeper, it appears at least `hide` is optional, refactored with https://github.com/yearn/yearn-sdk/pull/245/commits/b4de5d8312283e3c794546d395c3fd1970f89685

Reply:
Yes we should map the zapper response to our Token interface, and add additional info needed. We should filter out `hide === true` tokens, since they are not really supported, and have seen some scam tokens. We are filtering them in frontend currently, but we should have done it in sdk.

Reply:
Refactored to filter hidden tokens before mapping with https://github.com/yearn/yearn-sdk/pull/245/commits/55d449792283f354d58c682b6aa21a7bb8cadd61